src/components/new-safe/create/useSyncSafeCreationStep.ts

```diff
@@ -2,19 +2,20 @@ import { useEffect } from 'react'
 import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
 import type { NewSafeFormData } from '@/components/new-safe/create/index'
 import useWallet from '@/hooks/wallets/useWallet'
-import useIsWrongChain from '@/hooks/useIsWrongChain'
+import type { ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import { useCurrentChain } from '@/hooks/useChains'
 
-const useSyncSafeCreationStep = (setStep: StepRenderProps<NewSafeFormData>['setStep']) => {
+const useSyncSafeCreationStep = (setStep: StepRenderProps<NewSafeFormData>['setStep'], networks: ChainInfo[]) => {
   const wallet = useWallet()
-  const isWrongChain = useIsWrongChain()
+  const currentChain = useCurrentChain()
 
   useEffect(() => {
-    // Jump to choose name and network step if the wallet is connected to the wrong chain and there is no pending Safe
-    if (!wallet || isWrongChain) {
+    // Jump to choose name and network step if there is no pending Safe or if the selected network does not match the connected network
+    if (!wallet || (networks.length === 1 && currentChain?.chainId !== networks[0].chainId)) {
```

Comment:
Not sure if we should add this length 1 limitation or not. Currently, it is possible to select more than one network, go to the next step and switch to a network that is not selected and nothing happens. I think it would me more intuitive if it jumped back to the network selection in this case so that the user can add the current network to the selected networks as well. Wdyt?

Reply:
I'm not sure we even need the network selector during safe creation anymore. There is another PR that changes the network selector while you have a safe open to just show other chains in the multichain safe group, and additional chains you can deploy to, so the purpose of the network selector has changed a bit.

But before you've created a safe it doesn't really serve any purpose. If you want to change the networks you're deploying to I think the intuitive thing to do would be to go back to that step using the back button. And for multichain safes it doesn't matter what the currently selected network is.

The only thing the current network is used for now is if you're deploying a single chain safe, it will check the current network (which should be aligned with the network to be deployed) against the wallet network and show a warning if they're different. but we could just check the network to be deployed directly instead, and then the current network wouldn't be needed at all.

@schmanu wdyt?

Reply:
I agree that the network selector in the header on non-safe routes could be removed now. Probably makes sense to discuss it in #4124.

In the context of safe creation the behaviour as I understand it:

- If more than one network is selected there is only the pay later option for which the wallet network doesn't matter. If the user however changes the network through the header we ignore whatever they select. My suggestion: Remove the header network selector as part of #4124 

- If only one network is selected and if the user wants to pay now the wallet network matters as they have to execute a transaction. Since they might also have the option to pay later there is no reason to jump back to step 1 but instead show an info box and let them switch the network before submitting so this is in line with the current implementation.